tests/js/sentry-test/performance/utils.ts

```diff
@@ -0,0 +1,169 @@
+    offenderSpanIds: [],
+    parentSpanIds: [],
+  };
+
+  constructor() {
+    this._event = {
+      id: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+      eventID: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+      title: EXAMPLE_TRANSACTION_TITLE,
+      type: EventOrGroupType.TRANSACTION,
+      startTimestamp: 0,
+      endTimestamp: 0,
+      contexts: {
+        trace: {
+          trace_id: this.TRACE_ID,
+          span_id: this.ROOT_SPAN_ID,
+          op: 'pageload',
+          status: 'ok',
+          type: 'trace',
+        },
+      },
+      entries: [
+        {
+          data: this._spans,
+          type: EntryType.SPANS,
+        },
+      ],
+      perfProblem: {
+        causeSpanIds: [],
+        offenderSpanIds: [],
+        parentSpanIds: [],
+      },
+      // For the purpose of mock data, we don't care as much about the properties below.
+      // They're here to satisfy the type constraints, but in the future if we need actual values here
+      // for testing purposes, we can add methods on the builder to set them.
+      crashFile: null,
+      culprit: '',
+      dateReceived: '',
+      dist: null,
+      errors: [],
+      fingerprints: [],
+      location: null,
+      message: '',
+      metadata: {
+        current_level: undefined,
+        current_tree_label: undefined,
+        directive: undefined,
+        display_title_with_tree_label: undefined,
+        filename: undefined,
+        finest_tree_label: undefined,
+        function: undefined,
+        message: undefined,
+        origin: undefined,
+        stripped_crash: undefined,
+        title: undefined,
+        type: undefined,
+        uri: undefined,
+        value: undefined,
+      },
+      projectID: '',
+      size: 0,
+      tags: [],
+      user: null,
+    };
+  }
+
+  /**
+   *
+   * @param opts.startTimestamp
+   * @param opts.endTimestamp
+   * @param opts.op The operation of the span
+   * @param opts.description The description of the span
+   * @param opts.status Optional span specific status, defaults to 'ok'
+   * @param opts.numSpans If provided, will create the same span numSpan times
+   * @param opts.problemSpan If this span should be part of a performance problem, indicates the type of problem span (i.e ProblemSpan.OFFENDER, ProblemSpan.PARENT)
+   * @param opts.parentSpanId When provided, will explicitly set this span's parent ID. If you are creating nested spans via `childOpts`, this will be handled automatically and you do not need to provide an ID.
+   * Defaults to the root span's ID.
+   * @param opts.childOpts An array containing options for direct children of the current span. Will create direct child spans for each set of options provided
+   */
+  addSpan(opts: AddSpanOpts) {
+    const {
+      startTimestamp,
+      endTimestamp,
+      op,
+      description,
+      status,
+      problemSpan,
+      parentSpanId,
+      childOpts,
+    } = opts;
+
+    if (!opts.numSpans) {
+      opts.numSpans = 1;
+    }
```

Comment:
I think you can do one of the following here,
```suggestion
    const {
      startTimestamp,
      endTimestamp,
      op,
      description,
      status,
      problemSpan,
      parentSpanId,
      childOpts,
    } = opts;

    opts.numSpans ??=1
```

```suggestion
    const {
      startTimestamp,
      endTimestamp,
      op,
      description,
      status,
      problemSpan,
      parentSpanId,
      childOpts,
      numSpans = 1,
    } = opts;
```

Reply:
Second option works!